### PR TITLE
Add error handling, simplify code

### PR DESCRIPTION
**Pull request by safinsingh**
Implements error handling for basically everything, simplifies some code.

### CODE REVIEW

File: command/blocklist.go [hunk 1]
```diff
@@ -69,8 +70,9 @@ func blocklistRemove(ctx *Context, pattern string) {
 		ctx.Reply(fmt.Sprintf("Couldn't find `%s` in the blocklist", pattern))
 		return
 	}
-	ctx.Session.ChannelMessageSend(ctx.Message.ChannelID, fmt.Sprintf("the pattern `%s` has been removed from the blocklist", pattern))
-
+	if _, err = ctx.Session.ChannelMessageSend(ctx.Message.ChannelID, fmt.Sprintf("the pattern `%s` has been removed from the blocklist", pattern)); err != nil {
```
Comment (buffet) [inline command/blocklist.go]: I think you can use ctx.Reply here, which has error handling built in.

File: command/blocklist.go [hunk 2]
```diff
@@ -79,5 +81,7 @@ func blocklistGet(ctx *Context) {
 		ctx.ReportError("Failed to fetch patterns", err)
 		return
 	}
-	ctx.Session.ChannelMessageSend(ctx.Message.ChannelID, fmt.Sprintf("Patterns in the blocklist:\n```\n%s\n```", strings.Join(patterns, "\n")))
+	if _, err = ctx.Session.ChannelMessageSend(ctx.Message.ChannelID, fmt.Sprintf("Patterns in the blocklist:\n```\n%s\n```", strings.Join(patterns, "\n"))); err != nil {
```
Comment (buffet) [inline command/blocklist.go]: ctx.Reply

File: command/dotfiles.go [hunk 3]
```diff
@@ -30,7 +31,9 @@ func dotfiles(ctx *Context, args []string) {
 			return
 		}
 
-		ctx.Session.ChannelMessageSend(ctx.Message.ChannelID, profile.Dotfiles)
+		if _, err := ctx.Session.ChannelMessageSend(ctx.Message.ChannelID, profile.Dotfiles); err != nil {
```
Comment (buffet) [inline command/dotfiles.go]: ctx.Reply

File: command/git.go [hunk 4]
```diff
@@ -31,7 +32,9 @@ func git(ctx *Context, args []string) {
 			return
 		}
 
-		ctx.Session.ChannelMessageSend(ctx.Message.ChannelID, profile.Git)
+		if _, err = ctx.Session.ChannelMessageSend(ctx.Message.ChannelID, profile.Git); err != nil {
```
Comment (buffet) [inline command/git.go]: ctx.Reply

File: command/modping.go [hunk 5]
```diff
@@ -44,5 +44,7 @@ func modping(ctx *Context, args []string) {
 	if reason != "" {
 		reasonText = " for reason: " + reason
 	}
-	ctx.Session.ChannelMessageSend(ctx.Message.ChannelID, ctx.Message.Author.Mention()+" pinged moderators "+strings.Join(mods, " ")+reasonText)
+	if _, err := ctx.Session.ChannelMessageSend(ctx.Message.ChannelID, ctx.Message.Author.Mention()+" pinged moderators "+strings.Join(mods, " ")+reasonText); err != nil {
```
Comment (buffet) [inline command/modping.go]: ctx.Reply

File: command/move.go [hunk 6]
```diff
@@ -34,5 +35,7 @@ func move(ctx *Context, args []string) {
 		return
 	}
 	redirect := fmt.Sprintf("<https://discord.com/channels/%s/%s/%s>", ctx.Message.GuildID, m.ChannelID, m.ID)
-	ctx.Session.ChannelMessageSend(ctx.Message.ChannelID, fmt.Sprintf("Continued at <#%s> - %s (%s)", target, mentionsString, redirect))
+	if _, err = ctx.Session.ChannelMessageSend(ctx.Message.ChannelID, fmt.Sprintf("Continued at <#%s> - %s (%s)", target, mentionsString, redirect)); err != nil {
```
Comment (buffet) [inline command/move.go]: ctx.Reply

File: command/mute.go [hunk 7]
```diff
@@ -59,5 +60,7 @@ func mute(ctx *Context, args []string) {
 	if reason != "" {
 		r = " with reason: " + reason
 	}
-	ctx.Session.ChannelMessageSend(ctx.Env.ChannelModlog, fmt.Sprintf("User <@%s> was muted by %s for %s%s.", user, ctx.Message.Author.Username, duration, r))
+	if _, err := ctx.Session.ChannelMessageSend(ctx.Env.ChannelModlog, fmt.Sprintf("User <@%s> was muted by %s for %s%s.", user, ctx.Message.Author.Username, duration, r)); err != nil {
```
Comment (buffet) [inline command/mute.go]: ctx.Reply

File: command/repo.go [hunk 8]
```diff
@@ -1,7 +1,11 @@
 package command
 
+import "log"
+
 const repoHelp = "Sends a link to the bot's repository."
 
 func repo(ctx *Context, args []string) {
-	ctx.Session.ChannelMessageSend(ctx.Message.ChannelID, "https://github.com/unixporn/trup")
+	if _, err := ctx.Session.ChannelMessageSend(ctx.Message.ChannelID, "https://github.com/unixporn/trup"); err != nil {
```
Comment (buffet) [inline command/repo.go]: ctx.Reply

File: main.go [hunk 9]
```diff
@@ -159,14 +157,20 @@ func messageCreate(s *discordgo.Session, m *discordgo.MessageCreate) {
...
+			}
+
 			ch, err := s.UserChannelCreate(m.Author.ID)
 			if err != nil {
 				log.Println("Failed to create user channel with " + m.Author.ID)
 				return
 			}
 
-			s.ChannelMessageSend(ch.ID, "Your showcase submission was detected to be invalid. If you wanna comment on a rice, use the #ricing-theming channel.\nIf this is a mistake, contact the moderators or open an issue on https://github.com/unixporn/trup")
+			if _, err = s.ChannelMessageSend(ch.ID, "Your showcase submission was detected to be invalid. If you wanna comment on a rice, use the #ricing-theming channel.\nIf this is a mistake, contact the moderators or open an issue on https://github.com/unixporn/trup"); err != nil {
```
Comment (buffet) [inline main.go]: ctx.Reply, the return is unnecessary

File: main.go [hunk 10]
```diff
@@ -217,7 +227,10 @@ func messageCreate(s *discordgo.Session, m *discordgo.MessageCreate) {
 		}
 	}
 	if mentionsBot {
-		s.ChannelMessageSend(m.ChannelID, m.Author.Mention()+" need help? Type `!help`")
+		if _, err := s.ChannelMessageSend(m.ChannelID, m.Author.Mention()+" need help? Type `!help`"); err != nil {
```
Comment (buffet) [inline main.go]: ctx.Reply, unnecessary reply

File: main.go [hunk 11]
```diff
@@ -320,10 +333,15 @@ func cleanupMutes(s *discordgo.Session) {
 		err = s.GuildMemberRoleRemove(m.GuildId, m.User, env.RoleMute)
 		if err != nil {
 			log.Printf("Failed to remove role %s\n", err)
-			s.ChannelMessageSend(env.ChannelModlog, fmt.Sprintf("Failed to remove role Mute from user <@%s>. Error: %s", m.User, err))
+			if _, nestedErr := s.ChannelMessageSend(env.ChannelModlog, fmt.Sprintf("Failed to remove role Mute from user <@%s>. Error: %s", m.User, err)); nestedErr != nil {
+				log.Println("Failed to send Mute role removal message: " + err.Error())
+				return
```
Comment (mlvzk) [inline main.go]:
```suggestion
```

**Review by mlvzk — CHANGES_REQUESTED** [review]

**Comment by mlvzk** [comment]
I don't like some of the returns because they could break functionality.

@elkowar could you take a look?

File: main.go (hunk 11 above)
Comment (safinsingh) [reply main.go]: Ah my bad, you're right; this isn't a fatal error. Changing this along with buffet's suggestions.

**Comment by safinsingh** [comment]
@buffet, @mlvzk: I implemented your guys' suggestions, thanks!
@elkowar: I manually merged master into my branch. Would it be possible for you to make sure I didn't overwrite anything from #99?

File: command/pfp.go [hunk 12]
```diff
@@ -15,17 +16,23 @@ const (
...
-		ctx.Session.ChannelMessageSendEmbed(ctx.Message.ChannelID, &discordgo.MessageEmbed{
+
+		if _, err := ctx.Session.ChannelMessageSendEmbed(ctx.Message.ChannelID, &discordgo.MessageEmbed{
 			Title: fmt.Sprintf("%s#%s's profile picture", user.Username, user.Discriminator),
 			Image: &discordgo.MessageEmbedImage{
 				URL: avatar,
 			},
-		})
+		}); err != nil {
+			return err
```
Comment (buffet) [inline command/pfp.go]: Probably log an error here (to console not chat).

**Review by buffet — APPROVED** [review]

**Comment by buffet** [comment]
Looks good, can you resolve the conflicts before I merge?

**Comment by safinsingh** [comment]
@buffet ready to go!

**Review by mlvzk — APPROVED** [review]

**Pull request merged**